src/Maybe.js

```diff
@@ -1,45 +1,82 @@
-module.exports = Maybe;
+var util = require('./internal/util.js');
 
 function Maybe(x) {
-    if (!(this instanceof Maybe)) {
-        return new Maybe(x);
-    }
+    throw new Error('Maybe cannot be directly instantiated.');
```

Comment:
`Just(null)` still feels pretty weird to me, but after the discussion over at fantasy-land, i am going in that direction.